src/figuresClasses.ts

```diff
@@ -1,11 +1,81 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: string;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
```

Comment:
[CODE STYLE] - When throwing errors, provide a detailed description in the error message. Instead of 'All sides must be greater than 0', you could say 'Triangle side lengths must be greater than 0'.